bus-mapping/src/circuit_input_builder.rs

```diff
@@ -452,17 +462,57 @@ pub struct CircuitInputStateRef<'a> {
 
 impl<'a> CircuitInputStateRef<'a> {
     /// Push an [`Operation`] into the [`OperationContainer`] with the next
-    /// [`GlobalCounter`] and then adds a reference to the stored operation
+    /// [`RWCounter`] and then adds a reference to the stored operation
     /// ([`OperationRef`]) inside the bus-mapping instance of the current
-    /// [`ExecStep`].  Then increase the block_ctx [`GlobalCounter`] by one.
+    /// [`ExecStep`].  Then increase the block_ctx [`RWCounter`] by one.
     pub fn push_op<T: Op>(&mut self, op: T) {
         let op_ref = self
             .block
             .container
-            .insert(Operation::new(self.block_ctx.gc.inc_pre(), op));
+            .insert(Operation::new(self.block_ctx.rwc.inc_pre(), op));
         self.step.bus_mapping_instance.push(op_ref);
     }
 
+    /// Push a [`MemoryOp`] into the [`OperationContainer`] with the next
+    /// [`RWCounter`] and `call_id`, and then adds a reference to
+    /// the stored operation ([`OperationRef`]) inside the bus-mapping
+    /// instance of the current [`ExecStep`].  Then increase the `block_ctx`
+    /// [`RWCounter`] by one.
+    pub fn push_memory_op(
+        &mut self,
+        rw: RW,
+        address: MemoryAddress,
+        value: u8,
+    ) {
+        let call_id = self.call().call_id;
```

Comment:
We might need to add memory read/write to caller/callee's memory, maybe in the future we need to implement `push_caller_memory_op` and `push_callee_memory_op`, or just use `push_op(MemoryOp { ... })` to set `call_id` manually.

Reply:
> We might need to add memory read/write to caller/callee's memory, maybe in the future we need to implement `push_caller_memory_op` and `push_callee_memory_op`, or just use `push_op(MemoryOp { ... })` to set `call_id` manually.

I like `push_caller_memory_op` and `push_callee_memory_op`!  We can add them once we need them :)